### PR TITLE
Handle "Message blocked by harmful links filter" error

### PR DESCRIPTION
**Pull request by granny**

### CODE REVIEW

File: jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java [hunk 1]
```diff
@@ -236,25 +241,47 @@ private Message updateMessage() throws IllegalStateException {
...
+                        full = URL_PATTERN.matcher(full).replaceAll("$1");
+                        return sendOrEditMessage(full, channel);
+                    }
+                    throw new RuntimeException(errorResponseException2);
                 }
+            } else if (errorResponseException.getErrorCode() == MESSAGE_BLOCKED_BY_HARMFUL_LINK_FILTER_ERROR_CODE) {
+                full = URL_PATTERN.matcher(full).replaceAll("$1");
+                return sendOrEditMessage(full, channel);
+            } else {
+                throw new RuntimeException(errorResponseException);
```
Comment (Vankka) [inline jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java]: Alternative retry that reduces a bit of code duplication (untested), thoughts?

```suggestion
        try {
            for (int i = 0; i < 2; i++) {
                try {
                    return sendOrEditMessage(full, channel);
                } catch (ErrorResponseException ex) {
                    if (i == 0 && getErrorResponse() == ErrorResponse.UNKNOWN_MESSAGE) {
                        currentMessage = null;
                        continue;
                    }
                    throw ex;
                }
            }
        } catch (ErrorResponseException ex) {
            if (ex.getErrorCode() == MESSAGE_BLOCKED_BY_HARMFUL_LINK_FILTER_ERROR_CODE) {
                full = URL_PATTERN.matcher(full).replaceAll("$1");
                return sendOrEditMessage(full, channel);
            } else {
                throw new RuntimeException(errorResponseException);
```
also is that last wrap in RuntimeException wanted?

File: jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java [hunk 2]
```diff
@@ -236,25 +241,47 @@ private Message updateMessage() throws IllegalStateException {
...
+            }
+        } catch (ExecutionException | InterruptedException e) {
             if (this.isInterruptedException(e)) return currentMessage;
+
+            Throwable cause = e.getCause();
+            if (cause instanceof ErrorResponseException) {
+
+
+                throw (ErrorResponseException) cause;
+            }
```
Comment (Vankka) [inline jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java]:
```suggestion
            if (cause instanceof ErrorResponseException) {
                throw (ErrorResponseException) cause;
            }
```

File: jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java (hunk 1 above)
Comment (granny) [reply jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java]: code suggestion LGTM. question answered in discord.

File: jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java (hunk 2 above)
Comment (granny) [reply jda4/src/main/java/me/scarsz/jdaappender/ChannelLoggingHandler.java]: Merged.

**Review by Vankka — APPROVED** [review]

**Pull request merged**